packages/components/src/dropdown-menu/dropdown-menu.tsx

```diff
@@ -207,6 +208,49 @@ export const CheckboxItem = forwardRef<
 
 CheckboxItem.displayName = DropdownMenu.CheckboxItem.displayName
 
+/**
+ * SwitchItem
+ */
+
+type SwitchItemProps = DropdownMenu.DropdownMenuCheckboxItemProps & {
+  icon?: IconElement
+  label: string
+  checked: boolean
+  onCheckedChange: (checked: boolean) => void
+  danger?: boolean
+}
+
+export const SwitchItem = forwardRef<
+  React.ElementRef<typeof DropdownMenu.CheckboxItem>,
+  SwitchItemProps
+>((props, ref) => {
+  const { label, icon, danger, ...itemProps } = props
+
+  const id = useId()
+
+  return (
+    <DropdownMenu.CheckboxItem
+      ref={ref}
+      className={itemStyles()}
+      {...itemProps}
+    >
+      {icon && (
+        <span className={iconStyles({ danger })}>{cloneElement(icon)}</span>
+      )}
+      <span className={labelStyles({ danger })}>{label}</span>
+      <Switch
+        id={id}
+        checked={itemProps.checked}
+        onCheckedChange={() => {
+          // handled by parent
+        }}
```

Comment:
I find it useful to require both checked and onCheckedChange to be passed by the consumer (by default, they are both optional) because, in the majority of cases, you will want to do this. This case is an exception, but providing an empty function still seemed like a good reminder to future readers.